File: js/advanced/5d_count_subarrays_with_sum_k.js
//return the no. of subarrays whose sum = k. 

var subarraySum = function(nums, k) {
    const prefixSum = new Map()
    prefixSum.set(0,1)
    let sum = 0, ans = 0
    for(let i=0; i<nums.length; i++){
        sum+= nums[i]
        if(prefixSum.has(sum-k)) ans+= prefixSum.get(sum-k)
        prefixSum.set(sum, (prefixSum.get(sum) || 0) + 1)
    }
    return ans
};

File: c++/advanced/5d_count_subarrays_with_sum_k.cpp
//return the no. of subarrays whose sum = k. 

int subarraySum(vector<int>& nums, int k) {
        int sum=0, ans=0;
        unordered_map<int,int> m;
        m[0] = 1;
        for(int i=0; i<nums.size(); i++){
            sum+=nums[i];
            int diff = sum-k;
            if(m.find(sum-k) != m.end()){
                ans+= m[sum-k];
            }
            m[sum]++;
        }
        return ans;
    }